packages/apollo-server-lambda/src/ApolloServer.ts

```diff
@@ -173,30 +180,86 @@ export class ApolloServer extends ApolloServerBase {
         }
       }
 
-      const callbackFilter: APIGatewayProxyCallback = (error, result) => {
-        callback(
-          error,
-          result && {
-            ...result,
-            headers: {
-              ...result.headers,
-              ...requestCorsHeaders,
+      const makeCallbackFilter = (doneFn: any): APIGatewayProxyCallback => {
+        return (error, result) => {
+          doneFn();
+          callback(
+            error,
+            result && {
+              ...result,
+              headers: {
+                ...result.headers,
+                ...requestCorsHeaders,
+              },
             },
-          },
-        );
+          );
+        };
       };
 
-      graphqlLambda(async () => {
-        // In a world where this `createHandler` was async, we might avoid this
-        // but since we don't want to introduce a breaking change to this API
-        // (by switching it to `async`), we'll leverage the
-        // `GraphQLServerOptions`, which are dynamically built on each request,
-        // to `await` the `promiseWillStart` which we kicked off at the top of
-        // this method to ensure that it runs to completion (which is part of
-        // its contract) prior to processing the request.
-        await promiseWillStart;
-        return this.createGraphQLServerOptions(event, context);
-      })(event, context, callbackFilter);
+      const response = new Stream.Writable();
+
+      fileUploadProcess(event, response, this.uploadsConfig || {})
+        .then((body: any) => {
+          event.body = body;
+          graphqlLambda(async () => {
+            // In a world where this `createHandler` was async, we might avoid this
+            // but since we don't want to introduce a breaking change to this API
+            // (by switching it to `async`), we'll leverage the
+            // `GraphQLServerOptions`, which are dynamically built on each request,
+            // to `await` the `promiseWillStart` which we kicked off at the top of
+            // this method to ensure that it runs to completion (which is part of
+            // its contract) prior to processing the request.
+            await promiseWillStart;
+            return this.createGraphQLServerOptions(event, context);
+          })(
+            event,
+            context,
+            makeCallbackFilter(() => {
+              response.end();
+            }),
+          );
+        })
+        .catch(error => {
+          throw formatApolloErrors([error], {
+            formatter: this.requestOptions.formatError,
+            debug: this.requestOptions.debug,
+          });
+        });
     };
   }
 }
+
+const fileUploadProcess = (
```

Comment:
This name is too close to `processFileUploads`.  This might be better named as a `processRequest`, but that's not even quite right.  I realize that this code left some to be desired before, but I'm afraid that this makes it quite difficult to understand.  There are several other frameworks to use as a basis, but this is pretty far from all of them right now.